test/unit/model/scope.test.ts

```diff
@@ -0,0 +1,198 @@
+}
+
+const tempScope2: Scope = {
+  consentId: '1234',
+  action: 'balance',
+  accountId: 'sjdn-q333-2123'
+}
+
+/*
+ * Scope Resource Model Unit Tests
+ */
+describe('src/model/scope', (): void => {
+  let Db: Knex
+  let scopeDB: ScopeDB
+
+  beforeAll(async (): Promise<void> => {
+    Db = Knex(Config.test)
+    await Db.migrate.latest()
+    // Enable Sqlite foreign key support
+    await Db.raw('PRAGMA foreign_keys = ON')
+
+    scopeDB = new ScopeDB(Db)
+  })
+
+  afterAll(async (): Promise<void> => {
+    Db.destroy()
+  })
+
+  // Reset table for new test
+  beforeEach(async (): Promise<void> => {
+    await Db<Consent>('Consent').del()
+    await Db<Scope>('Scope').del()
+  })
+
+  describe('register', (): void => {
+    it('adds scope for an existing consent to the database', async (): Promise<void> => {
+      // Setup
+      await Db<Consent>('Consent')
+        .insert(partialConsent)
+
+      await scopeDB.register(tempScope)
+
+      // Assertion
+      const scopes: Scope[] = await Db<Scope>('Scope')
+        .select('*')
+        .where({
+          consentId: partialConsent.id
+        })
+
+      expect(scopes.length).toEqual(1)
+      expect(scopes[0].id).toEqual(expect.any(Number))
+      expect(scopes[0]).toEqual(expect.objectContaining(tempScope))
+    })
+
+    it('adds multiple scopes for an existing consent to the database', async (): Promise<void> => {
+      // Setup
+      await Db<Consent>('Consent')
+        .insert(partialConsent)
+
+      await scopeDB.register([tempScope, tempScope2])
+
+      // Assertion
+      const scopes: Scope[] = await Db<Scope>('Scope')
+        .select('*')
+        .where({
+          consentId: partialConsent.id
+        })
+
+      expect(scopes.length).toEqual(2)
+      expect(scopes[0].consentId === scopes[1].consentId).toEqual(true)
+      expect(scopes[0].action !== scopes[1].action).toEqual(true)
+    })
+
+    it('returns an error on adding a scope for a non-existent consent', async (): Promise<void> => {
+      await expect(scopeDB.register(tempScope)).rejects.toThrowError()
+    })
+  })
+
+  describe('retrieveAll', (): void => {
+    it('retrieves only existing scopes from the database', async (): Promise<void> => {
+      // Setup
+      await Db<Consent>('Consent')
+        .insert(partialConsent)
+      await Db<Scope>('Scope')
+        .insert(tempScope)
+
+      // Action
+      const scopes: Scope[] = await scopeDB.retrieveAll(tempScope.consentId)
+
+      // Assertion
+      expect(scopes.length).toEqual(1)
+      expect(scopes[0].id).toEqual(expect.any(Number))
+      expect(scopes[0]).toEqual(expect.objectContaining(tempScope))
+    })
+
+    it('throws an error on retrieving of non-existent scopes', async (): Promise<void> => {
+      // Setup
+      await Db<Consent>('Consent')
+        .insert(partialConsent)
+
+      // Action
+      await expect(scopeDB.retrieveAll(partialConsent.id))
+        .rejects.toThrowError('NotFoundError: Scope for ConsentId 1234')
```

Comment:
We should be able to test just for "NotFoundError" (the type) rather than the exact error message. Error messages tend to change over time, so we don't want to anchor on those.